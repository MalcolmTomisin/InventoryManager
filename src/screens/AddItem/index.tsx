import React from 'react';
import {ScrollView} from 'react-native';
import {TextInput} from '../../components';
import styles from './style';

export default function AddItem() {
  return (
    <ScrollView contentContainerStyle={styles.container}>
      <TextInput
        label="Name"
        placeholder="Tomisin Alu"
        containerStyle={styles.input}
          />
        
    </ScrollView>
  );
}
