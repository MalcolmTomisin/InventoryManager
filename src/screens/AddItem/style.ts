import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  input: {
    marginVertical: 10,
  },
});

export default styles;
