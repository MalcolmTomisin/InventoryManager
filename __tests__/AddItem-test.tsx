import 'react-native';
import React from 'react';
import AddItem from '../src/screens/AddItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const createTestProps = (props: Object) => ({
  navigation: {
    state: {params: {}},
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn(),
  },
  ...props,
});

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return {inventory: [], setOfNames: new Set()};
    }),
    useDispatch: jest.fn().mockImplementation(() => {
      return () => {};
    }),
  };
});

let props = createTestProps({});

it('renders correctly', () => {
  renderer.create(<AddItem {...props} />);
});
