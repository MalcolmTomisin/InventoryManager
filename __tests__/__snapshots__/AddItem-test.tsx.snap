// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
      "height": 1334,
      "paddingStart": 15,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "paddingVertical": 5,
            "width": 637.5,
          },
          Object {
            "marginVertical": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Aileron-Bold",
            "fontSize": 16,
          }
        }
      >
        Name
      </Text>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#9393AA",
            },
            Object {
              "borderBottomWidth": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          onEndEditing={[Function]}
          onFocus={[Function]}
          placeholder="Tomisin Alu"
          placeholderTextColor="#9393AA"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "borderWidth": 0,
                "color": "#272755",
                "fontFamily": "Aileron-Regular",
                "fontSize": 14,
              },
              Object {
                "height": 48,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingVertical": 5,
            "width": 637.5,
          },
          Object {
            "marginVertical": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Aileron-Bold",
            "fontSize": 16,
          }
        }
      >
        Price
      </Text>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#9393AA",
            },
            Object {
              "borderBottomWidth": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          onChangeText={[Function]}
          onEndEditing={[Function]}
          onFocus={[Function]}
          placeholder="â‚¦ 500"
          placeholderTextColor="#9393AA"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "borderWidth": 0,
                "color": "#272755",
                "fontFamily": "Aileron-Regular",
                "fontSize": 14,
              },
              Object {
                "height": 48,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingVertical": 5,
            "width": 637.5,
          },
          Object {
            "marginVertical": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Aileron-Bold",
            "fontSize": 16,
          }
        }
      >
        Total stock (Qty)
      </Text>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#9393AA",
            },
            Object {
              "borderBottomWidth": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          onChangeText={[Function]}
          onEndEditing={[Function]}
          onFocus={[Function]}
          placeholder="65"
          placeholderTextColor="#9393AA"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "borderWidth": 0,
                "color": "#272755",
                "fontFamily": "Aileron-Regular",
                "fontSize": 14,
              },
              Object {
                "height": 48,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingVertical": 5,
            "width": 637.5,
          },
          Object {
            "marginVertical": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Aileron-Bold",
            "fontSize": 16,
          }
        }
      >
        Description
      </Text>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#9393AA",
            },
            Object {
              "borderBottomWidth": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          onEndEditing={[Function]}
          onFocus={[Function]}
          placeholder="..."
          placeholderTextColor="#9393AA"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "borderWidth": 0,
                "color": "#272755",
                "fontFamily": "Aileron-Regular",
                "fontSize": 14,
              },
              Object {
                "height": 80,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#9393AA",
          "borderRadius": 5,
          "height": 48,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "GothamMedium",
            "fontSize": 12,
          }
        }
      >
        Add
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
